src/models/pathResolver.js

```diff
@@ -93,6 +93,9 @@ function updatePathWhenMapMoveEnd(location, map, router) {
   }${router.query.embed ? `&embed=true` : ''}`;
   if (router.query.tree_id) {
     result += `&tree_id=${router.query.tree_id}`;
+  } else if (router.query.walletid) {
+    const tree_id = router.asPath.match(/tree_id=(\d+)/)[1];
+    result += `&tree_id=${tree_id}`;
   }
```

Comment:
@henlam1 thanks for taking time to read code and do a PR, this is not so easy, given the fact that our complicated code.

Your code seems a way to handle this spesicifc case: walletid, but ideally, it would be better if we can handle this in a more general way, that means, we don't need to check the walletid, instead, we can detect all the parameter and put it into the path, does this make sense?

Reply:
Bye the way, we use jest test to unit test this model, so please run it locally to make should it's okay, also, if you can write some unit test to cover your new code, it would be great, like, pass in the URL of our case (URL with wallet and treeid), and check the result in your test

Reply:
https://github.com/Greenstand/treetracker-web-map-client/blob/main/src/models/pathResolver.test.js this is the test file

Reply:
@dadiorchen thanks for reviewing my PR. 

I agree it's better to handle this in a more general way and this issue is difficult because of the wallet URL structure. 
Selecting a tree produces different URLs depending on the query. For example,

- Planters: `planters/xxxx/trees/xxxx?bounds=`
- Organizations: `organizations/xxxx/trees/xxxx?bounds=`
- Wallets: `wallets/walletId/tokens?tree_id=xxxx` or `wallets/walletId/tokens?bounds=`

I'm not familiar with the wallet routing so I'd like to clarify and understand if the wallet URLs need to use `tree_id=xxxx` instead of `trees/xxxx`

Reply:
And thanks for letting me know about the unit tests

Reply:
I can not understand the question, for a general solution, you don't care the specific path (/trees or /wallet) you simply copy all those parameters (query param) into the new one

Reply:
@dadiorchen What I meant to ask is why does /wallet need a query to store the tree id? For /trees and /wallet, the tree id is stored in the pathname of the Location object.

Also I found a more general solution that passed the unit tests. Would greatly appreciate any feedback.

Reply:
Got you, yes, that's a special case, for wallet case, we can not use a token id, so we choose to work around by putting tree id here.